OutboxSample/Application/QueryHandlers/UserCommandQueryHandler.cs

```diff
@@ -0,0 +1,148 @@
+﻿using OutboxSample.Application.Commands;
+using OutboxSample.Application.DataAccess;
+using OutboxSample.Application.Eventing;
+using OutboxSample.Application.Queries;
+using OutboxSample.Common;
+using OutboxSample.DomainModel;
+using OutboxSample.DomainModel.Events;
+
+namespace OutboxSample.Application.QueryHandlers;
+
+public class UserCommandQueryHandler :
+    IQueryHandler<GetUserQuery, QueryResult<User>>,
+    IQueryHandler<GetAllUsersQuery, QueryResult<IEnumerable<DataStore.Item<User>>>>,
+    ICommandHandler<AddUserCommand, AddUserCommandResult>,
+    ICommandHandler<ChangeUserNameCommand, ChangeUserNameCommandResult>
```

Comment:
Would there be a QOL benefit to just using a traditional interface with defined methods like `GetUserAsync` or `ChangeUserAsync`, particularly if there would only ever be a single implementation in an application, instead of a series of `HandleAsync` methods (albeit with different parameter types)? It's not obvious when browsing the class what each `HandleAsync` method does without looking at the parameter types.
```csharp
public interface IUserCommandQueryHandler
{
  Task<QueryResult<User>> GetUserAsync(GetUserQuery query);
  Task<ChangeUserNameCommandResult> ChangeUserAsync(ChangeUserNameCommand command);
  ...
}
```

Reply:
Would there be a need for the same combination of `xCommand` and `xResult` be the same for other methods, or would this necessitate a new set of `yCommand` and `yResult` for each query or command?

Reply:
1. Interface with `GetUserAsync` API and etc vs Handler - I do not see big difference in _this_ specific case. For other. more complex scenarios this difference might matter. As in the 'traditional interface' case there has to be some single 'service' class. The concept of a handler, on the  other hand, is a bit more flexible - i.e. you can have a handler per command/query or, if you choose so, you can implement multiple handler interfaces in one class (as shown in the example. In some cases it might not make sense to put all the functionality into a single class because of low cohesion. In that case multiple separate handlers will look cleaner.
2. I'm not sure I fully understand the question on command/result  The general idea is that execution of a command/query results in smth. That 'smth' is represented as QueryResult/CommandResult. In a general case each distinct command/query will require its own Result.

Reply:
👍 , just confirming that each query/command would require a unique combination of `xCommand` and `yResult`